### PR TITLE
Fix for issue with customizable products returns

### LINKED ISSUE #9579: [BOOM-6042] You can return more customizable products than you ordered
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-6042).

- _**Reporter:**_ rdy4ever
- _**Created at:**_ Thu, 19 Jul 2018 19:21:32 +0200

<p>In order details, you can request a return for as many products as you ordered. But then, after the return is approved, or while is waiting for confirmation (or any other step in the order return process), you can still ask for a new return for the same order. Thus, the screenshot below: ordered 5 products but returned for 14.</p>

image
        
* How to reproduce the issue ?

Ask for a merchendise return, approve the return from bo (or not), go to the order details page and ask for the same return again. It still works.
<br/>
In the order details product table, the retuned quantity will be larger than the actual ordered quantity.

### PR DESCRIPTION
**Pull request by Prestaworks**
Remove all special code in template for customized order rows.
Add id_customization value on return detail for all rows.
This part of the code was having lots of legacy code from the time when customized order rows where connected to a single order row. Now when every new version of the customized product is a new order row, there is really no need to have special code for customized products as you can return these as any other order row. 
I made a small change to the OrderReturn class to keep id_customization on the the return_details table.

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc. Feel free to add more information below this table.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #9579.
| How to test?      | Make returns of customized products.
| Possible impacts? | Return handling

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26347)

### CODE REVIEW

File: classes/order/OrderReturn.php [hunk 1]
```diff
@@ -67,8 +67,10 @@ public function addReturnDetail($order_detail_list, $product_qty_list, $customiz
         /* Classic product return */
         if ($order_detail_list) {
             foreach ($order_detail_list as $key => $order_detail) {
+                $orderdetail = new OrderDetail((int) $order_detail);
+                $id_customization = $orderdetail->id_customization;
```
Comment (Progi1984) [inline classes/order/OrderReturn.php]: Could you move this code in the `if` ?

Comment (Prestaworks) [reply classes/order/OrderReturn.php]: Done!

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello, 
PR is tested on develop and the issue is fixed 
image
Many thanks ,

**Comment by Progi1984** [comment]
Thanks @Prestaworks & @marwachelly

**Pull request merged**